docs/lab/08/index.md

```diff
@@ -0,0 +1,435 @@
+- host identifier: is unique to the device in the network
+
+The *network mask* defines the number of bits that are used to represent the network prefix. In other words, the smaller the mask, the more unique devices can exist in the network.
+
+For example, for the above address `192.168.0.1/24`, we know that the network prefix is 24 bits long. Therefore, the first 24 bits give us the network prefix and the last 8 bits give us the host identifier. The mask in this case would look like this: `255.255.255.0`.
+
+:::info
+In binary, `255.255.255.0` is `11111111.11111111.11111111.00000000`. 
+
+First 24 bits are 1, last 8 bits are 0. 
+:::
+
+Any device that wants to communicate in this local network (or *subnet*) must have an address ranging from `192.168.0.1` to `192.168.0.254`.
+
+:::note
+In the above example, `192.168.0.0` is the *network address*, and `192.168.0.255` is the *broadcast address* (used to broadcast to *all* endpoints in a subnet). The first and last possible addresses in a subnet are reserved and should not be assigned to devices.
+
+To determine the *network address*, we need to perform a logic `AND` operation between the IP address and the network mask.
+
+To determine the *broadcast address*, we need to perform a logic `OR` operation between the IP address and the network mask.
+
+This means converting the IP address and network mask to binary, then performing `AND` or `OR` on every bit, then converting it back to decimal.
+:::
+
+Learn more about subnetting [here](https://networklessons.com/cisco/ccna-routing-switching-icnd1-100-105/what-is-subnetting).
+
+### Routers
+
+Routers are devices that forward packets between networks. The packets of the sender will be *routed* to the receiver of a different network, depending on the IP of the destination, passing from one router to another until they reach their destination. This happens through complex routing protocols. Routers also deal with traffic filtering, DNS (Domain Name System), DHCP and others.
+
+The *default gateway* in a local network is the router where packets must be sent when the *destination IP* is not in the current local network. From there, packets will be routed further. Read more about it [here](https://en.wikipedia.org/wiki/Default_gateway).
+
+### DHCP
+
+Dynamic Host Configuration Protocol (DHCP) is a network management protocol used by routers to dynamically allocate IP addresses to devices connected to its network. Routers dynamically assign addresses to new devices in the network, if they don't already have a *static* IP.
+
+### Networking protocols
+
+#### TCP (Transmission Control Protocol)
+
+TCP is *connection-oriented*, meaning that a connection must be first established between two devices (client and server) before data can be sent. Packets that are not received correctly must be *retransmitted* by the sender, ensuring that data always reaches its destination. TCP is used in cases where data integrity is most important.
```

Comment:
The packet might be fragmented during the transmission and the receiver might receive it in several smaller packets.